types/highland/index.d.ts

```diff
@@ -842,7 +906,7 @@ declare namespace Highland {
 		 * @param {Function} iterator - the function which reduces the values
 		 * @api public
 		 */
-		reduce1<U>(memo: U, f: (memo: U, x: R) => U): Stream<U>;
+		reduce1<U>(f: (memo: U, x: R) => U): Stream<U>;
```

Comment:
similar nitpick here, could we change the parameter name to be `iterator` to match the doc string

Reply:
Additionally, I think the type signature can be hairy for this -- on the first iteration, memo will be R, but can be anything after that depending on what the iterator returns. What do you think of this?

```
		reduce1<U>(iterator: (a: R | U, b: R) => U): Stream<U>;
```

One would expect that R=U most of the time, but it doesn't have to be. Propose these tests for this:

```
// $ExpectType Stream<Foo>
fooStream = fooStream.reduce1((memo: Foo, x: Foo) => {
  return memo;
});

// $ExpectType Stream<Bar>
barStream = fooStream.reduce1((memo: Foo | Bar, x: Foo) => {
  return bar;
});
```

Reply:
That's a good point, I hadn't thought of that use case.